code/modules/mob/language/language.dm

```diff
@@ -180,6 +180,9 @@
 /datum/language/proc/handle_message_mode(var/message_mode)
 	return list(src, message_mode)
 
+/datum/language/proc/can_speak_special(mob/speaker)
```

Comment:
There should be a comment explaining what that is used for.
The name of the proc is somewhat confusing as it might indicate that it is used to "enable" languages as opposed to "disabling" them.

It might be good to rename the proc to something like "check_restrict_speak" or something similar.

Reply:
renamed and comment added to clarify